src/lib.rs

```diff
@@ -52,6 +52,25 @@ pub struct DateTime {
     pub time: Time,
 }
 
+/// A duration, can hold three different formats.
+#[derive(Eq, PartialEq, Debug, Copy, Clone)]
+pub enum Duration {
+    /// consists of year, month, day, hour, minute and second units
```

Comment:
Not happy with our docs here, but they are bad on `Date` as well, so I'm fine with moving any changes here to a followup.

Reply:
I also can see some good methods to be implemented on Duration, but also in a followup.

Reply:
Alright, sounds good. 👍